### PR TITLE
DATA-69-Implement panoptic segmentation unit tests

### PR DESCRIPTION
**Pull request by VascoDVRodrigues**
## What?
 
This PR adds unit tests for the panoptic segmentation functionality. These tests ensure that the segmentation model performance metrics output correct results and handle edge cases.
 
## Why?
 
These changes are necessary to improve the reliability and robustness of the panoptic segmentation module. 
 
## How?
 
Implemented test functions to test `utils.py` and `pq.py`.

File | Function/Class | Test
-- | -- | --
utils.py | multiple_masks_to_single_mask | Empty prediction
utils.py | multiple_masks_to_single_mask | Single prediction
utils.py | multiple_masks_to_single_mask | No label2id dict
utils.py | multiple_masks_to_single_mask | Multiple non-overlapping masks
utils.py | multiple_masks_to_single_mask | Multiple overlapping masks
utils.py | payload_to_seg_metric | Mock payload
utils.py | payload_to_seg_metric | Empty payload
utils.py | payload_to_seg_metric | Missing model name
utils.py | payload_to_seg_metric | Missing gt field name
utils.py | payload_to_seg_metric | Already populated label2id
utils.py | payload_to_seg_metric | No provided label2id
utils.py | payload_to_seg_metric | Single sequence with 1 frame
utils.py | payload_to_seg_metric | Single sequence with multiple frames
pq.py | PanopticQuality | Class initialization
pq.py | PanopticQuality | Areas getter
pq.py | PanopticQuality | Update with numpy array
pq.py | PanopticQuality | Update with torch tensor
pq.py | PanopticQuality | Compute metric
pq.py | PanopticQuality | Update and compute metric
pq.py | PanopticQuality | 3 Updates and compute vs 1 update and compute
pq.py | PanopticQuality | Compute with multiple area ranges
 
## Testing
 
Run the unit tests using:
`pytest`
Captura de ecrã 2024-10-22, às 12 07 00
As you can see all tests were passed, with high code coverage on the panoptic side.

### CODE REVIEW

File: tests/panoptic/test_panoptic_pq.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+
+def test_compute_metric():
+    pq, label2id = setup_panoptic_quality()
+
+    payload = generate_synthetic_payload(gt_config=[1, 2], model_config=[1, 0])
+    pred, gt, label2id = payload_to_seg_metric(
+        payload=payload, model_name="model", label2id=label2id
+    )
+    pred[pred == -1] = 0
```
Comment (franziska-denk) [inline tests/panoptic/test_panoptic_pq.py]: If you like, you could add a comment in line 98 & 99 explaining why this is necessary, so that the code is easily understandable for everyone :)

File: tests/panoptic/test_panoptic_utils.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+def test_payload_to_seg_metric_partial_gt_and_preds():
+    payload = generate_synthetic_payload(gt_config=[1, 2, 0], model_config=[1, 0, 2])
+    pred, gt, label2id = payload_to_seg_metric(payload=payload, model_name="model")
+    assert label2id == {'MOTORBOAT': 0, 'SPHERICAL_BUOY': 1}
+    assert pred.shape == (3, 512, 640, 2)
+    assert gt.shape == (3, 512, 640, 2)
+    assert (pred != gt).any()
+    assert pred[0,:,:,0].sum() == pred[0,:,:,1].sum() == -327630
+    assert pred[1,:,:,0].sum() == pred[1,:,:,1].sum() == -327680
+    assert pred[2,:,:,0].sum() == pred[2,:,:,1].sum() == -327560
```
Comment (franziska-denk) [inline tests/panoptic/test_panoptic_utils.py]: I do not get how this number 327560 is created (but it should be correct as the tests are already checked in the PR, no?). i would expect it to be 327680-85

File: tests/panoptic/test_panoptic_utils.py [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+    assert gt[0,:,:,0].sum() == gt[0,:,:,1].sum() == -327630
+    assert gt[1,:,:,0].sum() == gt[1,:,:,1].sum() == -327560
```
Comment (franziska-denk) [inline tests/panoptic/test_panoptic_utils.py]: same here

**Review by franziska-denk — COMMENTED** [review]
Hey! super detailed tests and very clean implementation imo :) i just have two minor comments

**Review by franziska-denk — APPROVED** [review]
Thanks for adding the proposed changes, approved !
